v2/observer/observer.go

```diff
@@ -110,6 +125,17 @@ func (o *observer) run(ctx context.Context) error {
 	// Wait for an error or a completed context
 	var err error
 	select {
+	// In essence, this comprises the Observer's "healthcheck" logic.
+	// Whenever we receive an error on this channel, we cancel the context and
+	// shut down.  E.g., if one loop fails, everything fails.
+	// This includes:
+	//   1. an error pinging the API server endpoint
+	//      (Observer <-> API comms)
+	//   2. TODO
+	//      (Observer <-> K8s comms)
+	//
+	// Note: Currently, errors updating or cleaning up worker or job statuses
+	//       are handled by o.errFn, which currently simply logs the error
```

Comment:
fwiw, I think that's fine. We already have a period ping of the API server, so any failure here is less likely to be "zomg the API server is down" and more likely to be a specific issue with the event at hand.